proposals/0000-codable-range.md

```diff
@@ -0,0 +1,104 @@
+# Add Codable conformance to Range types
+
+* Proposal: [SE-NNNN](NNNN-codable-range.md)
+* Authors: [Dale Buckley](https://github.com/dlbuckley)
+* Review Manager: TBD
+* Implementation: [apple/swift#19532](https://github.com/apple/swift/pull/19532)
+* Status: **Awaiting review**
+
+## Introduction
+
+[SE-0167](0167-swift-encoders.md) introduced `Codable` conformance for some types in the standard
+library, but not the `Range` family of types. This proposal adds that
+conformance.
+
+Swift-evolution thread: [Range conform to Codable](https://forums.swift.org/t/range-conform-to-codable/15552)
+
+## Motivation
+
+`Range` is a very useful type to have conform to `Codable`. A good usage example is a range being sent to/from a client/server to convey a range of time using `Date`, or a safe operating temperature range using `Measurement<UnitTemperature>`.
+
+## Proposed solution
+
+The following Standard Library range types will gain `Codable` conformance
+when their `Bound` is also `Codable`:
+
+ * `Range`
+ * `ClosedRange`
+ * `PartialRangeFrom`
+ * `PartialRangeThrough`
+ * `PartialRangeUpTo`
+
+## Effect on ABI stability, resilience, and source stability
+
+This is a purely additive change, and so has no impact.
+
+## Alternatives considered
+
+One area of concern mentioned during the discussion is that of
+potential data corruption. Consider this: if an application implements its own
+`Codable` conformance for `Range` or `ClosedRange` (which is, by the way, not
+recommended, as both the protocol and the type are defined in the standard
+library), there might be data permanently stored somewhere (in the database, in
+a JSON or PLIST file, etc.) which was serialized using that conformance. Unless
+the serialization format is exaclty the same as the one used in the standard
```

Comment:
```suggestion
the serialization format is exactly the same as the one used in the standard
```

Reply:
I am now looking for a spell-checking plugin for the editor I used. Thanks, @benrimmington.